src/Microsoft.SqlTools.ManagedBatchParser/ReliableConnection/RetryPolicyUtils.cs

```diff
@@ -357,6 +377,38 @@ public static void RaiseSchemaAmbientRetryMessage(RetryState retryState, int err
             }
         }
 
+        /// <summary>
+        /// Wait for SqlConnection to handle sleeping serverless instances (allows for them to wake up).
+        /// </summary>
+        public static SqlRetryLogicBaseProvider ServerlessWaitRetryLogicProvider()
+        {
+            var serverlessRetryLogic = new SqlRetryLogicOption
+            {
+                NumberOfTries = _serverlessMaxRetries,
+                MaxTimeInterval = TimeSpan.FromSeconds(_serverlessMaxIntervalTime),
+                DeltaTime = TimeSpan.FromSeconds(1),
+                TransientErrors = _retryableExtendedNetworkConnectivityError
+            };
+
+            var provider = SqlConfigurableRetryFactory.CreateFixedRetryProvider(serverlessRetryLogic);
+
+            provider.Retrying += (object s, SqlRetryingEventArgs e) =>
+            {
+                int attempts = e.RetryCount + 1;
+                Logger.Write(TraceEventType.Information, $"attempt {attempts} - current delay time:{e.Delay} \n");
+                if (e.Exceptions[e.Exceptions.Count - 1] is SqlException ex)
```

Comment:
What do you think of 

```csharp
Logger.Write(
    TraceEventType.Information,
    (e.Exceptions[e.Exceptions.Count - 1] is SqlException ex)
    : $"{ex.Number}-{ex.Message}{Environment.NewLine}"
    ? $"{e.Exceptions[e.Exceptions.Count - 1].Message}{Environment.NewLine}");
```
which avoids duplicating some code?

Reply:
We also have overloads for each of the logger message types, so you can just do `Logger.Information(...)` instead of Logger.Write